testsuite/forge/src/backend/k8s/swarm.rs

```diff
@@ -189,6 +192,102 @@ impl K8sSwarm {
         k8snode.start().await?; // actually start the node. if port-forward is enabled, this is when it gets its ephemeral port
         Ok((peer_id, k8snode))
     }
+
+    async fn ensure_chaos_experiments_active(&self) -> Result<()> {
+        let timeout_duration = Duration::from_secs(30);
+        let polling_interval = Duration::from_secs(5);
+
+        tokio::time::timeout(timeout_duration, async {
+            loop {
+                match self.are_chaos_experiments_active().await {
+                    Ok(true) => {
+                        info!("Chaos experiments are active");
+                        return anyhow::Ok(());
+                    },
+                    Ok(false) => {
+                        info!("Chaos experiments are not active, retrying...");
+                    },
+                    Err(e) => {
+                        warn!(
+                            "Error while checking chaos experiments status: {}. Retrying...",
+                            e
+                        );
+                    },
+                }
+                tokio::time::sleep(polling_interval).await;
+            }
+        })
+        .await?
+        .map_err(|_| anyhow!("Timed out waiting for chaos experiments to be active"))
```

Comment:
`map_err` becomes a no-op because you have a `?` next to `.await` above, which short-circuits and returns any errors. I would suggest removing the `?` and keep the map_err and also wrap the inner error, like below.

```suggestion
        .map_err(|e| anyhow!("Timed out waiting for chaos experiments to be active: {}", e))
```